File: paleo_workbench/ui/pages/text_preview_widget.py
from __future__ import annotations

from PySide6.QtWidgets import QTextEdit

from paleo_workbench.ui import tokens


class TextPreviewWidget(QTextEdit):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setReadOnly(True)
        self.setLineWrapMode(QTextEdit.LineWrapMode.NoWrap)
        # 等宽字体族统一走 FONT_FAMILY_MONO token（主题无关，静态即可）。
        self.setStyleSheet(f"font-family: {tokens.FONT_FAMILY_MONO};")

    def load_text(self, text: str) -> None:
        self.setPlainText(text)

    def apply_settings(self, settings) -> None:
        font = self.font()
        font.setPointSize(settings.font_size)
        self.setFont(font)
        self.setStyleSheet(
            f"font-family: {tokens.FONT_FAMILY_MONO}; font-size: {settings.font_size}pt;"
        )
        self.setLineWrapMode(
            QTextEdit.LineWrapMode.WidgetWidth
            if settings.wrap_text
            else QTextEdit.LineWrapMode.NoWrap
        )

File: libs/ui_pages_preview/src/qt/text_preview_widget.cpp
#include <pwb/ui_pages_preview/qt/text_preview_widget.hpp>

#include <pwb/ui_pages_preview/preview_settings.hpp>

#include "../qt/style_util.hpp"

namespace pwb::ui_pages_preview {

TextPreviewWidget::TextPreviewWidget(QWidget* parent)
    : QTextEdit(parent) {
    setReadOnly(true);
    setLineWrapMode(QTextEdit::NoWrap);
    // Monospace family comes from the FONT_FAMILY_MONO token (theme-agnostic).
    setStyleSheet(QString("font-family: %1;").arg(qt_internal::FONT_FAMILY_MONO));
}

void TextPreviewWidget::apply_settings(const PreviewSettings& settings) {
    QFont f = font();
    f.setPointSize(settings.font_size);
    setFont(f);
    setStyleSheet(QString("font-family: %1; font-size: %2pt;")
                      .arg(qt_internal::FONT_FAMILY_MONO)
                      .arg(settings.font_size));
    setLineWrapMode(settings.wrap_text ? QTextEdit::WidgetWidth
                                       : QTextEdit::NoWrap);
}

}  // namespace pwb::ui_pages_preview
